src/WebJobs.Extensions.DurableTask/DurableTaskOptions.cs

```diff
@@ -252,6 +253,19 @@ internal void Validate()
                 throw new InvalidOperationException($"A non-empty {nameof(this.HubName)} configuration is required.");
             }
 
+            try
+            {
+                NameValidator.ValidateBlobName(this.HubName);
+                NameValidator.ValidateContainerName(this.HubName.ToLowerInvariant());
+                NameValidator.ValidateTableName(this.HubName);
+                NameValidator.ValidateQueueName(this.HubName.ToLowerInvariant());
+            }
+            catch (ArgumentException e)
+            {
+                throw new ArgumentException(
+                    $"Task hub name '{this.HubName}' can not be used for creation of Azure Storage resources - blobs, containers, tables or queues.", e);
```

Comment:
I would instead have a more generic error messages which explains the rules (only alpha-numeric, and must start with a letter - basically whatever the Azure Storage documentation says) rather than talking about Azure Storage constraints directly. That will give us more flexibility in the future if we change the underlying implementation.